File: ProyectoMysql/server/EntityLayer/OrdenCompraEntity.py
from datetime import datetime
from pydantic import BaseModel
from Utilidades.Enumerado.ProcessActionEnum import ProcessActionEnum


class OrdenCompraDetalleSaveModel (BaseModel):
    OrdenCompraDetalleId: int = 0
    OrdenCompraId: int = 0
    MercaderiaId: int = 0
    UnidadMedidaId: int = 0
    CantidadSolicitado: float = 0
    CantidadComprado: float = 0
    CantidadFaltante: float = 0
    PrecioUnitario: float = 0
    NomProducto: str = ''
    CategoriaId: int = 0
    CodigoUM: str = ''
    Stock: float = 0

    @classmethod
    def Cargar(cls, _DB):
        return cls.parse_obj(_DB)
    
class OrdenCompraSaveModel(BaseModel):
    OrdenCompraId: int = 0
    ProcesoId: int = 0
    EstadoProcesoId: int = 0
    TipoProcesoId: int = 0
    Codigo: str = ''
    EntidadId: int = 0
    NumDocumentoProveedor: str = ''
    NomProveedor: str = ''
    FechaEmision: datetime = datetime.now()
    FechaRegistro: datetime = datetime.now()
    CodUsuario: str = ''
    Action: ProcessActionEnum = ProcessActionEnum.Loaded
    NomEstadoProceso: str = ''
    DetalleItems: list[OrdenCompraDetalleSaveModel] = []

    @classmethod
    def Cargar(cls, _DB):
        return cls.parse_obj(_DB)

class OrdenCompraMainModel(BaseModel):
    OrdenCompraId: int = 0
    Codigo: str  = ''
    NumDocumentoProveedor: str = ''
    NomProveedor: str = ''
    FechaEmision: datetime = datetime.now()
    FechaRegistro: datetime = datetime.now()
    CodUsuario: str = ''
    NomEstadoProceso: str = ''
    ValorEstadoProceso: int = 0

    @classmethod
    def Cargar(cls, _DB):
        return cls.parse_obj(_DB)

File: ProyectoMysql/Api/LogisticStorage/LogisticStorage.EntityLayer/OrdenCompraEntity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LogisticStorage.EntityLayer
{
    public partial class OrdenCompraEntity : BaseEntityObject
    {
        public OrdenCompraEntity()
        {
            this.OrdenCompraId = 0;
            this.ProcesoId = 0;
            this.EstadoProcesoId = 0;
            this.TipoProcesoId = 0;
            this.Codigo = String.Empty;
            this.EntidadId = 0;
            this.NumDocumentoProveedor = String.Empty;
            this.NomProveedor = String.Empty;
            this.FechaEmision = DateTime.MinValue;
            this.FechaRegistro = DateTime.MinValue;
            this.CodUsuario = String.Empty;
            this.Detalles = new List<OrdenCompraDetalleEntity>();
            this.NomEstadoProceso = String.Empty;
            this.ValorEstadoProceso = 0;
        }
        public Int32 OrdenCompraId { get; set; }
        public Int32 ProcesoId { get; set; }
        public Int32 EstadoProcesoId { get; set; }
        public Int32 TipoProcesoId { get; set; }
        public String Codigo { get; set; }
        public Int32 EntidadId { get; set; }
        public String NumDocumentoProveedor { get; set; }
        public String NomProveedor { get; set; }
        public DateTime FechaEmision { get; set; }
        public DateTime FechaRegistro { get; set; }
        public String CodUsuario { get; set; }
        public List<OrdenCompraDetalleEntity> Detalles { get; set; }

    }
}
